cmd/juju/commands/bootstrap.go

```diff
@@ -502,8 +502,13 @@ func (c *bootstrapCommand) Run(ctx *cmd.Context) (resultErr error) {
 		if resultErr != nil {
 			if c.KeepBrokenEnvironment {
 				ctx.Infof(`
-bootstrap failed but --keep-broken was specified so resources are not being destroyed.
-When you have finished diagnosing the problem, remember to clean up the failed controller.
+bootstrap failed but --keep-broken was specified. 
+This means that this controller is no longer registered on your client. However,
```

Comment:
This means that cloud resources are left behind, but not registered to your local client, as the controller was not successfully created. However, you should be able to ssh into the machine using the user ubuntu and their IP address for diagnosis and investigation.
When you are ready to clean up the failed controller, use your cloud console or equivalent CLI tools to terminate the instances and remove remaining resources.